vkconfig/override_settings.cpp

```diff
@@ -48,11 +48,20 @@ OverrideSettings::OverrideSettings()
             if (layer.isObject()) {
                 QJsonValue components = layer.toObject().value("component_layers");
                 if (components.isArray()) {
-                    for(const QJsonValue& component : components.toArray()) {
+                    for (const QJsonValue &component : components.toArray()) {
                         if (!component.isString()) {
                             continue;
                         }
-                        layers.append(component.toString());
+                        enabled_layers.append(component.toString());
+                    }
+                }
+                QJsonValue blacklist = layer.toObject().value("blacklisted_layers");
+                if (blacklist.isArray()) {
+                    for (const QJsonValue &disable : blacklist.toArray()) {
+                        if (!disable.isString()) {
```

Comment:
when will it happen that the object is not a string?

Reply:
That value shouldn't ever be anything other than a string. But since it would be theoretically possible for another program to use an override, I felt it was best to make the check just to be safe.